### PR TITLE
[SYCL][ESIMD][EMU] lsc_atomic support

### PR DESCRIPTION
**Pull request by dongkyunahn-intel**
- __esimd_lsc_xatomic_slm_0/1/2
- __esimd_lsc_xatomic_bti_0/1/2/
- __esimd_lsc_xatomic_stateless_0/1/2

### CODE REVIEW

**Comment by dongkyunahn-intel** [comment]
Unrelated failures
SYCL / Linux / HIP AMDGPU LLVM Test Suite
```
Failed Tests (1):
  SYCL :: DeviceLib/imf_simd_emulate_test.cpp
```
SYCL / Linux / CUDA LLVM Test Suite
```
Failed Tests (1):
  SYCL :: DeviceLib/imf_simd_emulate_test.cpp
```

File: sycl/include/sycl/ext/intel/esimd/detail/atomic_intrin.hpp [hunk 1]
```diff
@@ -22,4 +41,82 @@ template <typename Ty> Ty atomic_add_fetch(Ty *ptr, Ty val) {
 #endif
 }
 
+template <typename Ty> Ty atomic_sub_fetch(Ty *ptr, Ty val) {
+#ifdef _WIN32
+  // TODO: Windows will be supported soon
+  __ESIMD_UNSUPPORTED_ON_HOST;
+#else
+  return __atomic_sub_fetch(ptr, val, __ATOMIC_RELAXED);
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/esimd/detail/atomic_intrin.hpp:49]: I'd suggest to use __ATOMIC_SEQ_CST every where to be on the safe side, unless we can prove relaxed is OK here.

Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/esimd/detail/atomic_intrin.hpp]: Done.

**Comment by dongkyunahn-intel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1018

File: sycl/include/sycl/ext/intel/esimd/detail/atomic_intrin.hpp [hunk 2]
```diff
@@ -13,12 +13,109 @@
 
 // This function implements atomic update of pre-existing variable in the
 // absense of C++ 20's atomic_ref.
+
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/esimd/detail/atomic_intrin.hpp:16]: I missed that completely the first time, but we have a big problem - those functions (atomic_load, etc.) pollute the user namespace. Please wrap it into namespace sycl::ext::intel::esimd::detail

**Review by kbobrovs — CHANGES_REQUESTED** [review]

**Comment by kbobrovs** [comment]
please fix test failures

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**